src/js/models/StopDetails.js

```diff
@@ -40,12 +34,8 @@ StopDetails.prototype.fetch = function() {
     function retryAtMost(maxRetries) {
         requests.get(yqlURL, params)
             .then(this.parseResponse.bind(this))
-            .tap(function(Services) {
-                var tripCollections = Services.map(function(Service) {
-                    return new TripCollection(this.stopID(), Service);
-                }.bind(this));
-
-                this.tripCollections(tripCollections);
+            .tap(function(Runs) {
+                this.tripCollection = new TripCollection(this.stopID(), Runs);
```

Comment:
This isn't directly related to this PR, but I think we could simplify retryAtMost by moving the catch stuff into their own functions.

```js
    function retryAtMost(maxRetries) {
        requests.get(yqlURL, params)
            .then(this.parseResponse.bind(this))
            .tap(function(Runs) {
                this.tripCollection = new TripCollection(this.stopID(), Runs);
                deferred.resolve();
            }.bind(this))
            .catch(CapMetroAPIError, handleCapMetroAPIError.bind(this, maxRetries))
            .catch(function(err) {
                this.errorMsg(err);
                deferred.reject(err);
            }.bind(this))
            .finally(function() {
                this.loading(false);
            }.bind(this));
    }

    function handleCapMetroAPIError(maxRetries, err) {
        var msg = err.message + '. Retrying ' + maxRetries + ' more times';
        console.error(msg);
        this.errorMsg(msg);

        if (maxRetries > 0) {
            return retryAtMost.call(this, maxRetries - 1);
        }
        else {
            deferred.reject(CapMetroAPIError);
        }
    }

    retryAtMost.call(this, config.MAX_RETRIES);

    return deferred.promise;
};
```

Generally speaking, this retry stuff could use some rethinking.